engine/queue/score-queue-impl/src/main/java/org/openscore/engine/queue/services/ExecutionQueueServiceImpl.java

```diff
@@ -73,6 +73,15 @@ public void enqueue(List<ExecutionMessage> messages) {
 				stateMessages.add(msg);
 			}
 		}
+
+        if (CollectionUtils.isNotEmpty(listeners)) {
+            long t = System.currentTimeMillis();
+            for (QueueListener listener : listeners) {
+                listener.prePersist(messages);
```

Comment:
Are you running all listeners here? don't you need specific ones for the prePresist?